app/src/app.rs

```diff
@@ -114,6 +117,34 @@ impl Component for App {
 
         let candidates = Rc::new(candidates);
 
+        let local_storage = match StorageService::new(Area::Local) {
+            Ok(storage) => Some(storage),
+            Err(e) => {
+                crate::log!("Local storage not supported: {:?}", e);
+                None
+            }
+        };
+
+        let candidates_selection = local_storage
+            .as_ref()
+            .and_then(|storage| {
+                match storage
+                    .restore::<Result<String, anyhow::Error>>("candidates-selection")
+                    .and_then(|x| {
+                        serde_json::from_str::<HashMap<String, CandidateStatus>>(&x)
+                            .context("could not deserialize")
+                    }) {
```

Comment:
Now I think the way this API is made you should use it like this... (I'm not sure):

```suggestion
                match storage
                    .restore::<yew::Format::Json<HashMap<String, CandidateStatus>>>("candidates-selection") {
```
(See how serde_json::from_str disappeared because it's included by yew::Format::Json)

Reply:
Didn't seems to work. 
```console
the trait `From<Result<std::string::String, anyhow::Error>>` is not implemented for `Result<Json<HashMap<std::string::String, CandidateStatus>>, anyhow::Error>`
```
It's look like `storage.restore() -> Result<String, Error>` but we construct a `Json::HashMap<String, CandidateStatus>`

Reply:
I tinkered with it and finally found how this API is supposed to be used.

All I did is check the doc: https://docs.rs/yew/0.18.0/yew/format/struct.Json.html

I noticed the Deserialize was not on Text but on `Result<something>`:

![image](https://user-images.githubusercontent.com/3018448/132823934-8bb90347-59ec-4b4a-ac3f-7db840ca1986.png)

Which I can match with the doc of restore (for which the reading is a bit obscured by the type alias):

![image](https://user-images.githubusercontent.com/3018448/132824405-b367d67c-6c39-4845-8132-d073891ba28a.png)

So `T` is actually `T: From<Result<String, Error>>;`. (As I always say: avoid type alias unless you have a very long type, otherwise it's obscuring the code (and the doc!))

You can now recognize that it matches the 3rd impl in the doc of `Json` (first screenshot).

So what we need to provide is something like `Json<Result<T, Error>>`.

`T` is the type that we want to deserialize from, it's our hashmap.

I was tempted to give anyhow::Error to the Result but if you look at the color of `Error` in the doc (1st screenshot) you will see it's actually a struct.  So it is `anyhow::Error` **forcibly**. That is  why it can be inferred and I ended up with `Json<Result<HashMap<String, CandidateStatus>, _>>`.


```rust
        let candidates_selection = local_storage
            .as_ref()
            .and_then(|storage| {
                use yew::format::Json;
                match storage
                    .restore::<Json<Result<HashMap<String, CandidateStatus>, _>>>(
                        "candidates-selection",
                    ) {
                    Json(Ok(map)) => Some(map),
                    Json(Err(err)) => {
                        crate::log!("Cannot restore data from local storage: {:?}", err);
                        None
                    }
                }
            })
            .unwrap_or_default();
```